### PR TITLE
Support kubectl alpha debug default image imagePullPolicy set by the image tag

### LINKED ISSUE #92383: imagePullPolicy does not default to Always for :latest images with ephemeral containers
**What happened**:
`kubectl alpha debug` reused my previous `:latest` image that is not pullable anymore.
**What you expected to happen**:
`kubectl alpha debug` to always pull the latest `:latest` image.  
As per documentation [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#ephemeralcontainer-v1-core), excerpt:
```
imagePullPolicystring | Image  pull policy. 
One of Always, Never, IfNotPresent. 
Defaults to Always if  :latest tag is specified, or IfNotPresent otherwise. 
Cannot be updated.  
More info:  https://kubernetes.io/docs/concepts/containers/images#updating-images
```
**How to reproduce it (as minimally and precisely as possible)**:
Have a simple Dockerfile with ENTRYPOINT ["/bin/bash"]
```
docker build my_image .
docker push my_image:latest
kubectl alpha debug -ti my_pod --image=my_image:latest --target=my_container
```
And then, change one line in your image,
```
docker build my_image .
docker push my_image:latest
kubectl alpha debug -ti my_pod --image=my_image:latest --target=my_container
kubectl describe pod my_pod
```
You will see that both images have the same sha256 (the first one built) and the Events in will show that it did not pull the image:
`Normal  Pulled   20m   kubelet, node005  Container image "my_image:latest" already present on machine
`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.8", GitCommit:"ec6eb119b81be488b030e849b9e64fda4caaf33c", GitTreeState:"clean", BuildDate:"2020-03-12T20:52:22Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

Regards,
Thank you for your time.

### PR DESCRIPTION
**Pull request by wawa0210**
**What type of PR is this?**

/kind bug
/kind feature

**What this PR does / why we need it**:

The default imagePullPolicy of kubectl alpha debug is PullIfNotPresent rather than set according to whether the image tag is `:latest`. 
This will cause the latest image to be executed without pull, if there is already a `:latest` image locally

**Which issue(s) this PR fixes**:

Fixes #92383

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Ephemeral containers now apply the same API defaults as initContainers and containers
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by wawa0210** [comment]
@verb 
Can you help me review this pr, thank you very much

**Comment by verb** [comment]
@wawa0210 yup, happy to review. Taking a look now.

File: pkg/apis/core/v1/defaults.go [hunk 1]
```diff
@@ -96,6 +96,20 @@ func SetDefaults_Container(obj *v1.Container) {
 	}
 }
 
+func SetDefaults_EphemeralContainer(obj *v1.EphemeralContainer) {
+	if obj.ImagePullPolicy == "" {
```
Comment (verb) [inline pkg/apis/core/v1/defaults.go:100]: I think we want the same defaulting for EphemeralContainer as for Container, could you instead invoke `SetDefaults_Container()` on `EphemeralContainerCommon`?

```
SetDefaults_Container((*v1.Container)(&obj.EphemeralContainerCommon))
```

File: pkg/apis/core/validation/validation.go [hunk 2]
```diff
@@ -2663,10 +2663,8 @@ func validatePullPolicy(policy core.PullPolicy, fldPath *field.Path) field.Error
 	allErrors := field.ErrorList{}
 
 	switch policy {
-	case core.PullAlways, core.PullIfNotPresent, core.PullNever:
+	case core.PullAlways, core.PullIfNotPresent, core.PullNever, "":
```
Comment (verb) [inline pkg/apis/core/validation/validation.go:2666]: Does this make policy an optional field? Was that intentional?

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go [hunk 3]
```diff
@@ -160,7 +160,7 @@ func addDebugFlags(cmd *cobra.Command, opt *DebugOptions) {
 	cmd.Flags().StringToString("env", nil, i18n.T("Environment variables to set in the container."))
 	cmd.Flags().StringVar(&opt.Image, "image", opt.Image, i18n.T("Container image to use for debug container."))
 	cmd.MarkFlagRequired("image")
-	cmd.Flags().String("image-pull-policy", string(corev1.PullIfNotPresent), i18n.T("The image pull policy for the container."))
+	cmd.Flags().String("image-pull-policy", "", i18n.T("The image pull policy for the container."))
```
Comment (verb) [inline staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:163]: Could you add a message to the description about the default behavior?

Comment (wawa0210) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: done.

File: pkg/apis/core/validation/validation.go (hunk 2 above)
Comment (wawa0210) [reply pkg/apis/core/validation/validation.go]: I misunderstood the order of container creation. In fact, this check happens after `SetDefaults_EphemeralContainer`

**Review by verb — APPROVED** [review]
Thanks for the fix!

/lgtm
/retest

**Review by zhouya0 — APPROVED** [review]
/lgtm

**Comment by wawa0210** [comment]
@liggitt @dims 

Can you help review this pr?

**Pull request merged**